internal/agent.go

```diff
@@ -74,9 +75,23 @@ func (a *Agent) HandleSPOE(ctx context.Context, writer *encoding.ActionWriter, m
 	app := a.Applications[appName]
 	a.mtx.RUnlock()
 	if app == nil {
-		// If we cannot resolve the app, we fail as this is an invalid configuration.
-		a.Logger.Panic().Str("app", appName).Msg("app not found")
-		return
+		if a.DefaultApplication == "" {
```

Comment:
What I would do differently is to check if the default app exists on config load and not storing a string but a pointer to said config. That way we can just check if app == nil && default app != nil, set it to default app and write the debug string.